agent-process-launcher/src/com/thoughtworks/go/agent/AgentOutputAppender.java

```diff
@@ -75,7 +75,7 @@ private RollingFileAppender rollingAppender(String file) throws IOException {
     }
 
     private static String getEffectiveLogDirectory(String file) {
-        String logDir = System.getenv(LOG_DIR);
+        String logDir = new SystemEnvironment().getPropertyImpl("go.agent.log.dir");
```

Comment:
That's not correct. `System.getenv(LOG_DIR)` is not the same as `System.getProperty('go.agent.log.dir')`. agent.sh has `LOG_DIR` set to logs or '/var/log/go-agent/'